### PR TITLE
install FILE_SET HEADERS for the new CMake versions 

### PR DESCRIPTION
**Pull request by ClausKlein**
this fix https://github.com/aminya/project_options/issues/143

### CODE REVIEW

File: src/PackageProject.cmake [hunk 1]
```diff
@@ -154,7 +154,9 @@ function(package_project)
     LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}" COMPONENT shlib
     ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}" COMPONENT lib
     RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}" COMPONENT bin
-    PUBLIC_HEADER DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/${_PackageProject_NAME}" COMPONENT dev)
+    PUBLIC_HEADER DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/${_PackageProject_NAME}" COMPONENT dev
+    FILE_SET HEADERS # NOTE: requierd from CMake to install the FILE_SET HEADERS too!
```
Comment (aminya) [inline src/PackageProject.cmake]:
```suggestion
    FILE_SET HEADERS # required by Cmake
```

File: src/PackageProject.cmake [hunk 2]
```diff
@@ -154,7 +154,9 @@ function(package_project)
     LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}" COMPONENT shlib
     ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}" COMPONENT lib
     RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}" COMPONENT bin
-    PUBLIC_HEADER DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/${_PackageProject_NAME}" COMPONENT dev)
+    PUBLIC_HEADER DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/${_PackageProject_NAME}" COMPONENT dev
+    FILE_SET HEADERS # NOTE: required from CMake to install the FILE_SET HEADERS too!
```
Comment (aminya) [inline src/PackageProject.cmake:158]: It fails on some of the CI configurations.

**Review by aminya — CHANGES_REQUESTED** [review]

File: src/PackageProject.cmake (hunk 2 above)
Comment (aminya) [reply src/PackageProject.cmake:158]: On which version, CMake has started to require this argument?

File: src/PackageProject.cmake [hunk 3]
```diff
@@ -148,13 +148,22 @@ function(package_project)
...
     TARGETS ${_targets_list}
     EXPORT ${_PackageProject_EXPORT}
     LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}" COMPONENT shlib
     ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}" COMPONENT lib
     RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}" COMPONENT bin
-    PUBLIC_HEADER DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/${_PackageProject_NAME}" COMPONENT dev)
+    PUBLIC_HEADER
+      DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/${_PackageProject_NAME}"
+      COMPONENT dev
+      ${FILE_SET_ARGS})
```
Comment (aminya) [inline src/PackageProject.cmake:166]: Changed the argument to this to only pass it on CMake 3.23 and more

Comment (ClausKlein) [reply src/PackageProject.cmake:166]: yes, it works

File: src/PackageProject.cmake (hunk 2 above)
Comment (ClausKlein) [reply src/PackageProject.cmake]: the [HEADER_SET](https://cmake.org/cmake/help/v3.24/prop_tgt/HEADER_SET.html?highlight=file_set) is new in `CMake v3.23`

**Pull request merged**